cryptopan/src/cryptopan.rs

```diff
@@ -0,0 +1,225 @@
+        let mut aes = Aes128EcbEnc::new(GenericArray::from_slice(&key[0..16]));
+
+        let mut buf = [0u8; 16];
+        buf.copy_from_slice(&key[16..32]);
+        let g = GenericArray::from_mut_slice(&mut buf);
+        aes.encrypt_block_mut(g);
+
+        Self {
+            aes,
+            pad: BitVec::from_slice(g.as_slice()),
+        }
+    }
+    /// Creates new outer block for encryption using str key. Panics if key size isn't 256 bits.
+    pub fn from_str_key(string_key: &str) -> Self {
+        Self::from_key(string_key.as_bytes())
+    }
+    /// Encrypts either a V4 or V6 IP address using the key
+    pub fn encrypt(&self, input: IpAddr) -> IpAddr {
+        let bits = match input {
+            V4(b) => BitVec::<u8, Msb0>::from_slice(&b.octets()),
+            V6(b) => BitVec::<u8, Msb0>::from_slice(&b.octets()),
+        };
+
+        let mut encrypted = bitvec!(u8, Msb0; 0; bits.len());
+        let mut padding = self.pad.clone();
+
+        // The first bit does not take from the original address
+        let mut encpadding = padding.clone();
+        let g = GenericArray::from_mut_slice(encpadding.as_raw_mut_slice());
+        self.aes.clone().encrypt_block_mut(g);
+        let firstpadding: BitVec<u8, Msb0> = BitVec::from_slice(g.as_slice());
+        encrypted.set(0, firstpadding[0] ^ bits[0]);
+
+        for n in 1..bits.len() {
+            padding.set(n - 1, bits[n - 1]);
+
+            // encrypt padded - this is used as a psuedorandom function
+            let mut encpadding = padding.clone();
+            let g = GenericArray::from_mut_slice(encpadding.as_raw_mut_slice());
+            self.aes.clone().encrypt_block_mut(g);
+            let res: BitVec<u8, Msb0> = BitVec::from_slice(g.as_slice());
+
+            // get the first bit of the encrypted value
+            encrypted.set(n, res[0] ^ bits[n]);
+        }
+
+        let bs = encrypted.as_bitslice();
+
+        if input.is_ipv4() {
+            V4(Ipv4Addr::new(
+                bs[0..8].load::<u8>(),
+                bs[8..16].load::<u8>(),
+                bs[16..24].load::<u8>(),
+                bs[24..32].load::<u8>(),
+            ))
+        } else {
+            V6(Ipv6Addr::new(
+                bs[0..16].load::<u16>(),
+                bs[16..32].load::<u16>(),
+                bs[32..48].load::<u16>(),
+                bs[48..64].load::<u16>(),
+                bs[64..80].load::<u16>(),
+                bs[80..96].load::<u16>(),
+                bs[96..112].load::<u16>(),
+                bs[112..128].load::<u16>(),
+            ))
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::{collections::HashMap, str::FromStr};
+
+    fn get_key() -> CryptoPAn {
+        CryptoPAn::from_key(&[
+            21u8, 34u8, 23u8, 141u8, 51u8, 164u8, 207u8, 128u8, 19u8, 10u8, 91u8, 22u8, 73u8,
+            144u8, 125u8, 16u8, 216u8, 152u8, 143u8, 131u8, 121u8, 121u8, 101u8, 39u8, 98u8, 87u8,
```

Comment:
You don't need to repeat `u8` for every element. Also I think it will be a bit more idiomatic to define a constant:
```rust
const KEY: [u8; 32] = [21, 34, ..];
```
You also could use the [`hex-literal`](https://docs.rs/hex-literal) crate, but I guess it's a bit redundant to bring it for just one constant.